From 1d75697d9b2ca4f431ebbe764400a4b97ab034df Mon Sep 17 00:00:00 2001
From: maxice8 <thinkabit.ukim@gmail.com>
Date: Fri, 5 Oct 2018 21:23:45 -0300
Subject: [PATCH] don't assume __linux__ means __GLIBC__ on BACKTRACES_ENABLE

check for __GLIBC__ instead of __linux__ as musl libc doesn't provide
execinfo.h

fixes compilation on Alpine Linux and Void Linux musl arches.
---
 programs/fileio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/programs/fileio.c b/programs/fileio.c
index ed3a29cda..1a73a75e4 100644
--- a/programs/fileio.c
+++ b/programs/fileio.c
@@ -21,7 +21,7 @@
 #endif
 
 #if !defined(BACKTRACES_ENABLE) && \
-   (defined(__linux__) || (defined(__APPLE__) && defined(__MACH__)) )
+   (defined(__GLIBC__) || (defined(__APPLE__) && defined(__MACH__)) )
 #  define BACKTRACES_ENABLE 1
 #endif
 
